pkg/config/dynamic/tcp_config.go

```diff
@@ -172,3 +173,17 @@ func (t *TCPServersTransport) SetDefaults() {
 	t.DialKeepAlive = ptypes.Duration(15 * time.Second)
 	t.TerminationDelay = ptypes.Duration(100 * time.Millisecond)
 }
+
+// +k8s:deepcopy-gen=true
+
+// TCPServer holds a TCP Server configuration.
+type TCPServerHealthCheck struct {
+	Address          string          `json:"address,omitempty" toml:"address,omitempty" yaml:"address,omitempty" label:"-"`
+	Port             string          `json:"-" toml:"-" yaml:"-"`
+	TLS              bool            `json:"tls,omitempty" toml:"tls,omitempty" yaml:"tls,omitempty"`
+	ServersTransport string          `json:"serversTransport,omitempty" toml:"serversTransport,omitempty" yaml:"serversTransport,omitempty" export:"true"`
+	Interval         ptypes.Duration `json:"interval,omitempty" toml:"interval,omitempty" yaml:"interval,omitempty" export:"true"`
```

Comment:
We could go a little further by adding an unhealthy interval to the TCP health check configuration. It has been added for HTTP health check by https://github.com/traefik/traefik/pull/10610.
WDYT?